src/components/FavoritesView/index.js

```diff
@@ -25,27 +61,8 @@ const SearchView = () => {
 	return(
 		<Wrapper>
 			<Container>
-				{!favorites.length ? (
-					<InfoScreen
-						emoji={<HeartBreak size={96} style={{margin: '1rem'}}/>}
-						title='You still have no favorites'
-						description='you can add movies to your favorites clicking on the ♥ icon'
-					/>
-				) : (
-					<Fragment>
-						<Text weight={600} xs={2} sm={3} md={4} xg={5}>{getTitle(favorites.length)}</Text>
-						<Row vertical-gutter style={{marginTop: '2rem', marginBottom: '2rem'}}>
-							{favorites.map(favorite => (
-								<Cell key={favorite} xs={6} sm={4} md={3} xg={2}>
-									<FetchCard
-										kindURL={favorite?.split('/')[0]}
-										id={favorite?.split('/')[1]}
-									/>
-								</Cell>
-							))}
-						</Row>
-					</Fragment>
-				)}
+				<HasFavorites favorites={favorites}></HasFavorites>
```

Comment:
```
<HasFavorites favorites={favorites}></HasFavorites>
<NoFavorites length={favorites.length}></NoFavorites>
```
Esses dois componentes, utilizam favorites e length, ambos para fazer a mesma coisa, um if para decidir se devem exibir o componente ou não. Neste caso, não deveriam os dois receberem ou favorites ou length?
Isso não viola o "principle of least surprise"?

Reply:
Faz todo sentido, foi algo que quando estava desenvolvendo me incomodou, mas estava sem ideias, usei timebox, mas não consegui enxergar algo, a ideia é sempre evoluir.